client/server/pages/index.js

```diff
@@ -52,20 +52,24 @@ const debug = debugFactory( 'calypso:pages' );
 
 const calypsoEnv = config( 'env_id' );
 
+let branchName;
 function getCurrentBranchName() {
-	try {
-		return execSync( 'git rev-parse --abbrev-ref HEAD' ).toString().replace( /\s/gm, '' );
-	} catch ( err ) {
-		return undefined;
+	if ( ! branchName ) {
+		try {
+			branchName = execSync( 'git rev-parse --abbrev-ref HEAD' ).toString().replace( /\s/gm, '' );
```

Comment:
These are drive-by local performance improvements. In dev mode, these will run on every request (and sometimes more frequently than that). `execSync` is relatively slow in this context, so caching the value while the server is online is a good way to speed things up.